src/engraving/dom/keysig.cpp

```diff
@@ -197,12 +198,39 @@ PointF KeySig::staffOffset() const
     return PointF(0.0, 0.0);
 }
 
+EngravingItem* KeySig::propertyDelegate(Pid propertyId)
+{
+    switch (propertyId) {
+    case Pid::KEY:
+    case Pid::KEY_CONCERT:
+    case Pid::SHOW_COURTESY:
+    case Pid::KEYSIG_MODE:
+    case Pid::IS_COURTESY: {
+        if (staff() && segment()) {
+            const Fraction changeTick = staff()->currentKeyTick(tick());
+            if (Segment* s = score()->tick2segment(changeTick, true, SegmentType::KeySig)) {
+                return toKeySig(s->element(staff2track(staffIdx())));
+            }
+        }
+        break;
+    }
+    default:
+        break;
+    }
+
+    return nullptr;
+}
+
 //---------------------------------------------------------
 //   getProperty
 //---------------------------------------------------------
 
 PropertyValue KeySig::getProperty(Pid propertyId) const
 {
+    EngravingItem* e = const_cast<KeySig*>(this)->propertyDelegate(propertyId);
```

Comment:
In order not to forget it: this needs a rebase after https://github.com/musescore/MuseScore/pull/30151